File: grind75/two_sum.go
package grind75

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, x := range nums {
		need := target - x
		if n, ok := m[need]; ok {
			return []int{i, n}
		}
		m[x] = i
	}
	return nil
}

File: src/main/java/neetcode150/array/TwoSum.java
package neetcode150;

import java.util.HashMap;
import java.util.Map;

// https://leetcode.com/problems/two-sum/
public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>(nums.length);
        for (int i = 0; i < nums.length; i++) {
            int cmp = target - nums[i];
            if (map.containsKey(cmp)) {
                return new int[]{i, map.get(cmp)};
            }
            map.put(nums[i], i);
        }
        return new int[]{0, 0};
    }
}
